pkg/loki/common/common.go

```diff
@@ -74,11 +76,11 @@ type Storage struct {
 }
 
 func (s *Storage) RegisterFlagsWithPrefix(prefix string, f *flag.FlagSet) {
-	s.S3.RegisterFlagsWithPrefix(prefix+".s3", f)
-	s.GCS.RegisterFlagsWithPrefix(prefix+".gcs", f)
-	s.Azure.RegisterFlagsWithPrefix(prefix+".azure", f)
-	s.Swift.RegisterFlagsWithPrefix(prefix+".swift", f)
-	s.BOS.RegisterFlagsWithPrefix(prefix+".bos", f)
+	s.S3.RegisterFlagsWithPrefix(prefix, f)
+	s.GCS.RegisterFlagsWithPrefix(prefix, f)
+	s.Azure.RegisterFlagsWithPrefix(prefix, f)
+	s.Swift.RegisterFlagsWithPrefix(prefix, f)
+	s.BOS.RegisterFlagsWithPrefix(prefix, f)
```

Comment:
This looks like a breaking change that needs extra mention in the backlog, right?

Reply:
Yes, this is a breaking change to the CLI flags, because these were registered incorrectly. Not sure what the best course of action is here 😕  See comment: https://github.com/grafana/loki/pull/7916#issuecomment-1348276269

Reply:
I think for the sake of BC, we should keep the old flags and put a deprecation notice in the upgrade guide, and create an issue for v3.0 to remove these deprecated flags.

Reply:
As discussed internally via slack, this is not a breaking change because the CLI flags with the prefix `common.storage` are not yet available to the default flag system. These are only set on the throwaway flag system to allow for common config application logic: https://github.com/grafana/loki/pull/7916/files#diff-5789ed7c47e1cf4394cef6ad3bc4aaea1af3552e930b0420703708e6bd7d2c89R54-R57